### PR TITLE
[ADD] ir_rule_website: new setting to bypass website rules when worki…

### PR DESCRIPTION
**Pull request by iledarn**
…ng from backend

### CODE REVIEW

**Comment by iledarn** [comment]
@yelizariev, done

File: ir_rule_website/README.rst [hunk 1]
```diff
@@ -8,12 +8,20 @@ Allows to use ``website_id`` (current website) in ``domain_force`` field of Reco
...
+Example of usage:
 
 * Show a blog on specific websites only (TODO: add link to the module)
 * Show an event on specific websites only (TODO: add link to the module)
 * Show a product on specific websites only (TODO: add link to the module)
 
+Known issues / Roadmap
+======================
+
+* There is no `webisite_id` attribute in a rule evaluation context when working from Odoo backend.
```
Comment (yelizariev) [inline ir_rule_website/README.rst]: Say explicitly what does it mean for users of the module

File: ir_rule_website/README.rst [hunk 2]
```diff
@@ -8,12 +8,20 @@ Allows to use ``website_id`` (current website) in ``domain_force`` field of Reco
...
+  In this module we've introduced a new setting in ``ir.rule`` model to bypass website_id-rules in backend.
```
Comment (yelizariev) [inline ir_rule_website/README.rst]: What this statement in Roadmap section? It must be in description part

File: ir_rule_website/doc/index.rst [hunk 3]
```diff
@@ -10,7 +10,29 @@ Installation
...
+* Note: if you leave this fields empty for a rule that is using ``website_id`` parameter - you may have this kind of error when trying to configure your product's **website_ids** field from odoo backend:
+
+::
+
+ The requested operation cannot be completed due to security restrictions. Please contact your system administrator.
+
+ (Document type: product.template, Operation: read)
+
+ This is so because in backend rules work in non-website context.
+ In other words the setting is mandatory for website rules.
```
Comment (yelizariev) [inline ir_rule_website/doc/index.rst]: Fix format 
format

File: ir_rule_website/models/ir_rule.py [hunk 4]
```diff
@@ -16,4 +25,44 @@ def _eval_context(self):
...
+        if not rule_ids:
+            return []
+
+        # browse user and rules as SUPERUSER_ID to avoid access errors!
+        eval_context = self._eval_context()
+        user_groups = self.env.user.groups_id
+        global_domains = []                     # list of domains
+        group_domains = []                      # list of domains
+        for rule in self.browse(rule_ids).sudo():
+            # begin redefined part of original _compute_domain of odoo/base/addons/ir/ir_rule.
```
Comment (yelizariev) [inline ir_rule_website/models/ir_rule.py]: Use upper case for some words to easily distinguish comment from original comment

**Comment by yelizariev** [comment]
This branch has conflicts that must be resolved

**Comment by iledarn** [comment]
@yelizariev , done

**Pull request merged**